vendor/bat-native-ledger/src/bat/ledger/internal/api/api_parameters.cc

```diff
@@ -97,23 +92,25 @@ void APIParameters::RunCallbacks() {
   }
 }
 
-void APIParameters::SetRefreshTimer(const int delay) {
-  if (refresh_timer_id_ != 0) {
+void APIParameters::SetRefreshTimer(
+    base::TimeDelta delay,
+    base::TimeDelta base_delay) {
+  if (refresh_timer_.IsRunning()) {
     BLOG(1, "Params timer in progress");
     return;
   }
 
-  int64_t start_in = delay;
-  if (delay == 0) {
-    const int64_t base_delay = 3 * base::Time::kSecondsPerHour;
-    const int64_t random_delay =
-        brave_base::random::Geometric(10 * base::Time::kSecondsPerMinute);
-    start_in = base_delay + random_delay;
-  }
+  base::TimeDelta start_in =
+      base_delay + braveledger_time_util::GetRandomizedDelay(delay);
+
+  BLOG(1, "Params timer set for " << start_in);
 
-  BLOG(1, "Params timer start in " << start_in);
+  refresh_timer_.Start(FROM_HERE, start_in,
+      base::BindOnce(&APIParameters::OnTimerElapsed, base::Unretained(this)));
```

Comment:
can we just call `Fetch` directly here so that we don't need another function `OnTimerElapsed`